src/main/java/uk/org/tombolo/DataExportRunner.java

```diff
@@ -1,113 +1,121 @@
 import uk.org.tombolo.execution.DataExportEngine;
-import uk.org.tombolo.recipe.DataExportRecipeValidator;
+import uk.org.tombolo.exporter.CSVExporter;
+import uk.org.tombolo.exporter.GeoJsonExporter;
 import uk.org.tombolo.importer.ImporterMatcher;
+import uk.org.tombolo.recipe.DataExportRecipe;
+import uk.org.tombolo.recipe.FieldRecipe;
 
-import java.io.*;
+import java.io.Writer;
+import java.util.List;
 import java.util.Properties;
 
 public class DataExportRunner extends AbstractRunner {
     private static final Logger log = LoggerFactory.getLogger(DataExportRunner.class);
-    public static String executionSpecPath = "";
+    private static final DataExportRunner runner = new DataExportRunner();
 
     public static void main(String[] args) throws Exception {
-        validateArguments(args);
+        runner.validateArguments(args);
 
-        executionSpecPath = args[0];
-        String outputFile = args[1];
-        String forceImports = args[2];
-        Boolean clearDatabaseCache = Boolean.parseBoolean(args[3]);
-        String executionSpec = args[4];
+        Boolean isString = Boolean.parseBoolean(args[0]);
+        String recipe = args[1];
+        String output = args[2];
+        String correlation = args[3];
+        String forceImports = args[4];
+        Boolean clearDatabaseCache = Boolean.parseBoolean(args[5]);
 
-        run(executionSpecPath, outputFile, forceImports, clearDatabaseCache, executionSpec);
-    }
+        runner.validateRecipe(recipe, isString);
 
-    protected static void run(String executionSpecPath, String outputFile,
-                              String forceImports, Boolean clearDatabaseCache) throws Exception {
-    
-        run(executionSpecPath, outputFile, forceImports, clearDatabaseCache, "None");
+        run(isString, recipe, output, correlation, forceImports, clearDatabaseCache);
     }
 
-    protected static void run(String executionSpecPath, String outputFile,
-                              String forceImports, Boolean clearDatabaseCache, String executionSpec) throws Exception {
+    private static void run(Boolean isString, String recipe, String output, String correlation, String forceImports,
+                            Boolean clearDatabaseCache) throws Exception {
         HibernateUtil.startup();
         if (clearDatabaseCache) {
             DatabaseUtils.clearAllData();
         }
 
         // Load API keys
-        Properties apiKeys = loadApiKeys();
-
-        // Create engine
-        DataExportEngine engine = new DataExportEngine(apiKeys, initialiseDowloadUtils());
+        Properties apiKeys = runner.loadApiKeys();
 
-        if (!executionSpecPath.equalsIgnoreCase("None")) validateSpecification(executionSpecPath);
-        if (!executionSpec.equalsIgnoreCase("None")) validateSpecificationOfString(executionSpec);
+        // Loading the recipe for future use
+        DataExportRecipe dataExportRecipe = runner.getRecipe(recipe, isString);
```

Comment:
We could put this in one step, if recipe is validated just get the recipe, if not exit

Reply:
We call validate recipe before calling this method, so it gets here only if it's valid. Would you like to have this immediately after?

Reply:
Just a thought to combine these steps as if the recipe is valid we need to get it anyways, why call another method to get it, if valid lets just get it. Not super urgent, just a thought